### PR TITLE
tsoa yargs dependency update

### PR DESCRIPTION
**Pull request by grahamplata**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**

Put #1097 (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**

**Test plan**

### CODE REVIEW

**Comment by WoH** [comment]
Have you checked if any breaking changes may affect tsoa?

File: packages/cli/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "typescript": "^4.1.2",
     "validator": "^13.6.0",
     "yamljs": "^0.3.0",
-    "yargs": "^14.0.0"
+    "yargs": "^15.3.0"
```
Comment (bill-titus) [inline packages/cli/package.json:40]: Not sure about compatibility, but 17.2.1 is the latest version of yargs.  15.3.0 is about 2 years old.  15.4.1 is about 1 year old.

Comment (grahamplata) [reply packages/cli/package.json:40]: I can make that adjustment to `15.4.1` let me validate

**Comment by WoH** [comment]
Can you please upgrade the respective types package aswell.

**Comment by grahamplata** [comment]
> Can you please upgrade the respective types package as well.

I can! I'll have that back shortly

**Pull request merged**